crates/nargo/src/resolver.rs

```diff
@@ -94,19 +95,19 @@ impl<'a> Resolver<'a> {
             let crate_id = self.driver.create_non_local_crate(entry_path, *crate_type);
             self.driver.add_dep(parent_crate, crate_id, dep_pkg_name);
 
-            self.cached_packages.insert(dir_path, (crate_id, dep_meta));
+            cached_packages.insert(dir_path, (crate_id, dep_meta));
         }
 
         // Resolve all transitive dependencies
-        for (dir_path, (crate_id, dep_meta)) in self.cached_packages.iter() {
+        for (dir_path, (crate_id, dep_meta)) in cached_packages.into_iter() {
```

Comment:
`into_iter` shouldn't be necessary here

Reply:
Why would we not want to use `into_iter` here?

`cached_packages` is about to go out of scope so why not pass ownership of the contained values to the new resolver/error rather than cloning?

Reply:
I think even without the `.into_iter` it would also take ownersip unless you use an `&` infront of `cached_packages`

Reply:
Ah, I misunderstood this to mean I should leave it as `cached_packages.iter()`. Made the change now.